### ISSUE TITLE
Fetch CLI help strings instead of manually rewrite them

### ISSUE THREAD

**Issue by gaspardcereza** [shimming-toolbox/fsleyes-plugin-shimming-toolbox#30]
### Description of Problem / Inefficiency:
To display the contextual help of our different functions in the plugin (when the user hovers the 'i' buttons), we manually rewrite the docstrings in the plugin code. When we change the docstrings, we therefore need to update both the shimming-toolbox and the plugin repos...

### Implementation Suggestion:
We could consider using something like:
```python
from shimmingtoolbox.cli.b1shim import b1shim_cli
b1shim_cli.params[0].help
```
that returns `'Complex 3D B1+ map.'` to fetch the help strings from the Click objects. This way, we wouldn't need to change the docstrings in both the shimming-toolbox and the plugin repo, making maintenance much less cumbersome.

@po09i @jcohenadad let me know if you see any cons in doing so.

**Comment by jcohenadad** [comment]
looks like a neat idea, i love it!

**Comment by gaspardcereza** [comment]
To do this, we would however need to install shimmingtoolbox in `pst_venv` to access the CLI. I don't think this would cause any problem if we keep running the st code in `st_venv` only. @po09i do you foresee any problem that this could cause ?

To be honest, I still don't fully get why we cannot create a single virtual environnement with fixed versions of `fsleyes` and all the libraries we use. It seems to me that this would make things easier to maintain.

**Comment by jcohenadad** [comment]
> To be honest, I still don't fully get why we cannot create a single virtual environnement with fixed versions of fsleyes and all the libraries we use. It seems to me that this would make things easier to maintain.

That seems like a good idea indeed. Any reason we currently don't do that @po09i ?

**Comment by po09i** [comment]
### Why 2 Venvs
> > To be honest, I still don't fully get why we cannot create a single virtual environnement with fixed versions of fsleyes and all the libraries we use. It seems to me that this would make things easier to maintain.
> 
> That seems like a good idea indeed. Any reason we currently don't do that @po09i ?

The problem with having a single virtual environment is that `Fsleyes` has its own dependencies which are different from Shimming Toolbox's own dependencies. This means that there are conflicts of versions if we try to install both in the same environment. This is the reason why there are 2 environments. 

### Implementation suggestions
That being said, the feature you are proposing @gaspardcereza is a great idea. It would require installing Shimming Toolbox in the `Fsleyes plugin package` without the "invalid" dependencies so that it can run simple tasks such as fetching the docstring, however, we should not run proper API calls to Shimming Toolbox. 

Note: This could lead to confusion in the future since ST would be installed but not "useable".

At first glance, I see a couple of ways we could start:
- I believe this is doable with something like `pip install "$HOME/shimming-toolbox/shimmin-toolbox-$VERSION"` and adding a flag to resolve the dependencies properly. Something like
  - `--no-deps`
  - `--upgrade-strategy xxxxx`
- I don't know if it is possible but conda has the exact flag we would need: `--no-update-deps` which does not touch the already installed dependencies
- Another solution is to install Shimming Toolbox first in `pst_venv` then install `Fsleyes` and forcing to reinstall the deps so that it ignores ST's deps. The options I'm thinking are:
  - --ignore-installed
  - --force-reinstall

TLDR: It seems possible to install ST in the `pst_venv` without affecting the required dependencies of `Fsleyes`, this would obviously mean that we can't use ST for API calls but this is not a problem for the intended use. It would allow to fetch the docstring automatically.

**Comment by gaspardcereza** [comment]
## Why 2 Venvs

But wouldn't it be possible to make ST compatible with a specific version of fsleyes so that they can be used in a same venv ? Do you know what dependencies are causing trouble ? I just remember that we had problem with the version of `dcm2niix ` used by FSLeyes being too old for ST but this was then solved by Paul McCarthy in his next release. Maybe I missed some other conflicts that you might have encountered here and there.

## Implementation suggestions

I admit that it might be confusing from the user point of view to have ST installed but unusable in `pst_venv`, but I doubt many of them will actual dig what packages are installed where. Especially since we used launchers that don't necessitate to play with the different conda environments.

**Comment by po09i** [comment]
> But wouldn't it be possible to make ST compatible with a specific version of fsleyes so that they can be used in a same venv ?

Probably (only SciPy looks like an issue according to both dependency list). However that seems like a big limitation for this single use case. We would be limited to having the "same" dependencies as Fsleyes which seems like a big constraint for future developments of ST. Also, we would be able to more easily update Fsleyes in the future to get their latest features. This essentially makes ST a Fsleyes add-on rather than its own Toolbox with a plugin to Fsleyes.

> I admit that it might be confusing from the user point of view

I meant from a developper's POV sorry if I was not clear. The user would not notice at all.

**Comment by gaspardcereza** [comment]
> This essentially makes ST a Fsleyes add-on rather than its own Toolbox with a plugin to Fsleyes.

I see you're point. That would indeed be a problem. 

> I meant from a developper's POV sorry if I was not clear. The user would not notice at all.

In that case we can simply add a commented line in the `pst_venv` installation files to explain why we install ST.
